### ISSUE TITLE
reduce size of live data

### ISSUE THREAD

**Issue by peterdudfield** [openclimatefix/MetOfficeDataHub#57]
## Detailed Description
Would be great to reduce the size of the NWP data. I think this could be done with a better compression.

## Context
- Current szie is 1.2 GB - https://s3.console.aws.amazon.com/s3/buckets/nowcasting-nwp-development?region=eu-west-1&prefix=data/&showversions=false
- currently saves as .netcdf file
- other compressions has been discusse here - https://github.com/openclimatefix/ocf_datapipes/issues/132#issuecomment-1398195182

## Possible Implementation
use `Blosc Zstd` compression

**Comment by jacobbieker** [comment]
I would try the new `ocf_blosc2` Blosc2 ZSTD one, as that does give better results than the original Blosc Zstd

**Comment by peterdudfield** [comment]
Just so it super easy to do, do you a link to some code, where you save it using `ocf_blosc2`,

**Comment by jacobbieker** [comment]
Yeah, it would be something like this, which I use for the new satellite zarrs:

```
from ocf_blosc2.ocf_blosc2 import Blosc2

def write_to_zarr(dataset, zarr_name, mode, chunks):
    mode_extra_kwargs = {
        "a": {"append_dim": "time"},
        "w": {
            "encoding": {
                "data": {
                    "compressor": Blosc2("zstd", clevel=5),
                },
                "time": {"units": "nanoseconds since 1970-01-01"},
            }
        },
    }
    extra_kwargs = mode_extra_kwargs[mode]
    dataset.chunk(chunks).to_zarr(
        zarr_name, compute=True, **extra_kwargs, consolidated=True, mode=mode
    )
```

**Comment by devsjc** [comment]
Update deployed NWP consumer:
- [x] dev via [Terraform cloud run](https://app.terraform.io/app/openclimatefix/workspaces/nowcasting_infrastructure_development-eu-west-1/runs/run-5EjE5gV9pZ55U5YK)

**Comment by devsjc** [comment]
Updated forecaster to include OCF blosc 2 library:
- [x] dev via [Terraform cloud run](https://app.terraform.io/app/openclimatefix/workspaces/nowcasting_infrastructure_development-eu-west-1/runs/run-mdF7ja16uXnhUGCD)
- [ ] prod

**Comment by peterdudfield** [comment]
great work @devsjc

**Comment by devsjc** [comment]
Not quite there yet, I'll let you know!

**Comment by devsjc** [comment]
Which repository holds the national model? I will need to update that to be able to read the newly compressed NWP data as well

**Comment by devsjc** [comment]
Seeing the following error in the forecaster in [cloudwatch](https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#logEventViewer:group=/aws/ecs/forecast/;stream=streaming/forecast/65216956eee84a1d97fc5bbf6a2cc631):

```sh
expected shape=(7, 24, 24, 11) actual shape (4, 24, 24, 11)
```

**Comment by peterdudfield** [comment]
I just rolled back nwp to `1.2.0` from `1.2.2` on devlopment as it was causing an issue on development

https://eu-west-1.console.aws.amazon.com/cloudwatch/home?region=eu-west-1#logsV2:log-groups/log-group/$252Faws$252Fecs$252Fconsumer$252Fnwp$252F/log-events/streaming$252Fnwp-consumer$252F52032316e23c4ca5b31f8a4aa527cd4c

![Screenshot 2023-05-27 at 08 01 49](https://github.com/openclimatefix/MetOfficeDataHub/assets/34686298/aab7eadf-cb91-4728-b878-f5da0c8712a8)
